src/main/java/io/openliberty/tools/intellij/runConfiguration/LibertyRunManagerListener.java

```diff
@@ -40,8 +41,11 @@ public void runConfigurationRemoved(@NotNull RunnerAndConfigurationSettings sett
             try {
                 VirtualFile vBuildFile = VfsUtil.findFile(Paths.get(runConfig.getBuildFile()), true);
                 LibertyModule libertyModule = libertyModules.getLibertyModule(vBuildFile);
-                if (libertyModule != null && libertyModule.getCustomStartParams().equals(runConfig.getParams())) {
-                    libertyModule.setCustomStartParams(""); // clear the custom params
+                if (libertyModule != null) {
+                    libertyModule.setRunInContainer(false); // clear the run in container checkbox
+                    if (libertyModule.getCustomStartParams().equals(runConfig.getParams())) {
```

Comment:
@aparnamichael do you know the reasoning behind this check? What is the reason we don't just go ahead and clear the custom params every time a run configuration is removed?

Reply:
@aparnamichael I was just reviewing with @TrevCraw and have similar questions.

Reply:
@mrglavas @TrevCraw These condition is there in the initial file commit. I also think that condition is not needed. But not sure whether that will introduce any new bugs.

Reply:
Looking into this further, it looks like the condition is to ensure that the run config being deleted is the one that was last run for this particular Liberty module. One Liberty module can have many run configs associated with it. If a user deletes a run config that was not the last one run, then we should not be resetting the Liberty module params to the default values. The same logic should apply for the "run in container" setting.

Reply:
As a side note, the current "check" to see if a config is last one run is not foolproof. It is possible configs could be identical. Configs could also have all the same custom dev mode params and have other differences (like the "run in container" setting). 

@mrglavas I think this just further shows a larger re-design is required, as you recommended in our discussion. 

@aparnamichael, as @turkeylurkey suggested in another comment, please open an issue for redesign. I would suggest a generic issue like "Redesign the relationship between Liberty module and run configurations". We can put the specific suggestion from @turkeylurkey in there, but @mrglavas recommended an even bigger overhaul of the design.

Reply:
@turkeylurkey and @TrevCraw created a new issue for re-design. https://github.com/OpenLiberty/liberty-tools-intellij/issues/977

Reply:
Added more detail to that issue.